### PR TITLE
Allow 1x1 dataframes to get blacklisted

### PR DESCRIPTION
**Pull request by peterhurford**

### CODE REVIEW

File: R/db.R [hunk 1]
```diff
@@ -284,11 +284,11 @@ write_data_safely <- function(dbconn, tblname, df, key, safe_columns, blacklist)
     if (length(blacklist) > 0L) {
       df_without_id <- df[-which(names(df) == key)]  # Don't blacklist on id.
       if (any(is.na(blacklist))) {
-        df <- df[apply(!is.na(df_without_id), 1, all),]
+        df <- df[apply(!is.na(df_without_id), 1, all), , drop = FALSE]
         blacklist <- blacklist[!is.na(blacklist)]
       }
       if (length(blacklist) > 0L) {
-        df <- df[apply(df_without_id, 1, function(x) all(!(x %in% blacklist))),]
+        df <- df[apply(df_without_id, 1, function(x) all(!(x %in% blacklist))), , drop = FALSE]
```
Comment (khalidharun) [inline R/db.R:291]: @robertzk I think there's a cool opportunity here for making this more DRY and using function composition to create a `filter_fn` and for the `apply(df_without_id, 1, filter_fn)`

Comment (peterhurford) [inline R/db.R:291]: Sounds like a good idea. We also need a more efficient df -> df function than `apply`. But I think we can do either thing in a future PR.

**Comment by khalidharun** [comment]
Besides the comment above, LGTM.

File: R/db.R (hunk 1 above)
Comment (robertzk) [inline R/db.R:291]: @khalidharun Agreed that we need a faster `apply(., 1, ...)`. Looking through plyr/dplyr doesn't yield anything at the moment -- the best I've found is http://stackoverflow.com/questions/13978918/apply-is-slow-how-to-make-it-faster-or-what-are-my-alternatives but it doesn't give us what we want.

**Pull request merged**